kernel/kernel-api/src/main/java/io/delta/kernel/internal/stats/FileSizeHistogram.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.stats;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.internal.data.GenericRow;
+import io.delta.kernel.internal.util.VectorUtils;
+import io.delta.kernel.types.ArrayType;
+import io.delta.kernel.types.LongType;
+import io.delta.kernel.types.StructType;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/** A histogram that tracks file size distributions and their counts. */
+public class FileSizeHistogram {
+
+  //////////////////////////////////
+  // Static variables and methods //
+  //////////////////////////////////
+
+  private static final long KB = 1024;
+  private static final long MB = KB * 1024;
+  private static final long GB = MB * 1024;
+
+  private static final List<Long> DEFAULT_BIN_BOUNDARIES = createDefaultBinBoundaries();
+
+  public static final StructType FULL_SCHEMA =
+      new StructType()
+          .add("sortedBinBoundaries", new ArrayType(LongType.LONG, false))
+          .add("fileCounts", new ArrayType(LongType.LONG, false))
+          .add("totalBytes", new ArrayType(LongType.LONG, false));
+
+  /**
+   * Creates default bin boundaries as specified by the Delta Lake specification. The boundaries
+   * are: - 0 and powers of 2 till 4 MB - 4 MB jumps till 40 MB - 8 MB jumps till 120 MB - 4 MB
+   * jumps till 144 MB - 16 MB jumps till 576 MB - 64 MB jumps till 1408 MB - 128 MB jumps till 2 GB
+   * - 256 MB jumps till 4 GB - power of 2 till 256 GB
+   */
+  private static List<Long> createDefaultBinBoundaries() {
+    List<Long> sizes = new ArrayList<>();
+
+    // 0 and powers of 2 till 4 MB
+    sizes.add(0L);
+    for (long size = 8 * KB; size <= 4 * MB; size *= 2) {
```

Comment:
btw -- why this change? I quite liked how it was before, it was very clear and explicit?

Reply:
Ah, I see the comment that suggested this. Sounds good to me :)